File: HRMS/HRMS.Web/Modules/ServerTypes/Recruitment/CandidatesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CandidatesRow } from "./CandidatesRow";
import { CandidateStatus } from "./CandidateStatus";

export interface CandidatesColumns {
    CandidateId: Column<CandidatesRow>;
    JobTitle: Column<CandidatesRow>;
    FirstName: Column<CandidatesRow>;
    LastName: Column<CandidatesRow>;
    Email: Column<CandidatesRow>;
    Mobile: Column<CandidatesRow>;
    ResumePath: Column<CandidatesRow>;
    Status: Column<CandidatesRow>;
    AppliedDate: Column<CandidatesRow>;
}

export class CandidatesColumns extends ColumnsBase<CandidatesRow> {
    static readonly columnsKey = 'Recruitment.Candidates';
    static readonly Fields = fieldsProxy<CandidatesColumns>();
}

[CandidateStatus]; // referenced types

File: HRMS/HRMS.Web/Modules/Recruitment/Candidates/CandidatesColumns.cs
namespace HRMS.Recruitment.Columns;

[ColumnsScript("Recruitment.Candidates")]
[BasedOnRow(typeof(CandidatesRow), CheckNames = true)]
public class CandidatesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int CandidateId { get; set; }
    public string JobTitle { get; set; }
    [EditLink]
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Email { get; set; }
    public string Mobile { get; set; }
    public string Resume { get; set; }
    [DisplayName("Progress Status")]
    public CandidateStatus Status { get; set; }
    public DateTime AppliedDate { get; set; }
}
